planet/clients/orders.py

```diff
@@ -324,8 +318,17 @@ async def download_order(
 
         Raises:
             planet.exceptions.APIException: On API error.
+            planet.exceptions.StateError: If the order is not in a completed
+                state.
         """
         order = await self.get_order(order_id)
+        if order.state not in ORDERS_STATES_COMPLETE:
+            raise exceptions.StateError(
```

Comment:
I feel like StateError is too specific. Maybe just OrderError again? Less is more with exception types, I think.

Reply:
My thinking on adding `StateError` is to indicate situations where a waiter might be useful. For an order, this is when it's not in a completed state. For an image in the Data API, this is when the image is not active. I am not as clear on where/if it happens in the other APIs. As for the less is more approach, it seems to make sense but I just haven't read enough about exception design to know for sure. Do you have any resources or examples that could provide context?

Reply:
just falling back to ClientError for now